cmd/kubeadm/app/phases/upgrade/versiongetter.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"fmt"
+	"io"
+
+	clientset "k8s.io/client-go/kubernetes"
+	versionutil "k8s.io/kubernetes/pkg/util/version"
+)
+
+// versionGetter defines an interface for fetching different versions.
+// Easy to implement a fake variant of this interface for unit testing
+type versionGetter interface {
+	GetClusterVersion() (string, *versionutil.Version, error)
```

Comment:
Same comments as before about the unnecessary `Get` prefix:

https://golang.org/doc/effective_go.html#Getters